internal/dao/sqlx/account_dao.go

```diff
@@ -23,83 +24,83 @@ type accountDaoSqlx struct {
 	list               *sqlx.Stmt
 }
 
-func getAccountDao(ctx context.Context, tx dao.Transaction) (dao.AccountDao, error) {
+func getAccountDao(ctx context.Context) (dao.AccountDao, error) {
 	var err error
 	daoImpl := accountDaoSqlx{}
 	daoImpl.name = "account"
 
-	daoImpl.getById, err = tx.PreparexContext(ctx, getAccountById)
+	daoImpl.getById, err = db.DB.PreparexContext(ctx, getAccountById)
 	if err != nil {
 		return nil, NewPrepareStatementError(ctx, &daoImpl, getAccountById, err)
 	}
-	daoImpl.getByAccountNumber, err = tx.PreparexContext(ctx, getAccountByAccountNumber)
+	daoImpl.getByAccountNumber, err = db.DB.PreparexContext(ctx, getAccountByAccountNumber)
 	if err != nil {
 		return nil, NewPrepareStatementError(ctx, &daoImpl, listAccounts, err)
 	}
-	daoImpl.getByOrgId, err = tx.PreparexContext(ctx, getAccountByOrgId)
+	daoImpl.getByOrgId, err = db.DB.PreparexContext(ctx, getAccountByOrgId)
 	if err != nil {
 		return nil, NewPrepareStatementError(ctx, &daoImpl, listAccounts, err)
 	}
-	daoImpl.list, err = tx.PreparexContext(ctx, listAccounts)
+	daoImpl.list, err = db.DB.PreparexContext(ctx, listAccounts)
 	if err != nil {
 		return nil, NewPrepareStatementError(ctx, &daoImpl, listAccounts, err)
 	}
 
 	return &daoImpl, nil
 }
 
-func (dao *accountDaoSqlx) NameForError() string {
-	return dao.name
+func (di *accountDaoSqlx) NameForError() string {
+	return di.name
```

Comment:
FYI I renamed all reciever variables to `di` because this was shadowing imported `dao` package.

I think what's ugly in Go is that the language allows variable shadowing. Superb talk on this topic: https://www.youtube.com/watch?v=zPd0Cxzsslk

Reply:
just watched it, should we start using `go vet -shadow` ? :)

Reply:
Totally a good idea, feel free to submit a patch, we probably have many already.